.github/workflows/docs.yml

```diff
@@ -43,7 +43,9 @@ jobs:
     - name: Test and build docs
       run: |
         python -m doctest README.rst
-        python setup.py build_sphinx
+        pushd docs
+        make html
```

Comment:
kinda confused on this change. the makefile just invokes the `setup.py build_sphinx` for us in the same way.

Reply:
@kratsg similar, but not exact, no? 

https://github.com/scikit-hep/pyhf/blob/1f152585232b919075189ea80eb42d531347609c/docs/Makefile#L54-L58

Reply:
Still not sure why we want to change to explicitly using `make html`. https://www.sphinx-doc.org/en/master/usage/advanced/setuptools.html seems to be supported more nicely  -- so I'm not sure.

Reply:
Let me revert this change here so that we can discuss this more in an issue. I could be wrong, and will need to go hunting for references that support this idea, that using `setuptools` to do anything explicitly is more and more being looked at as "bad" by the Python community and that APIs through more dedicated tooling is preferred. 

Off the top of my head I'm thinking of https://github.com/tox-dev/tox/issues/1100 as being a starting place for some of these discussions.

I have been wanting to move away from using `setuptools` for a while, and I would normally suggest using [`sphinx-build` directly](https://www.sphinx-doc.org/en/1.8/man/sphinx-build.html) but 

https://github.com/scikit-hep/pyhf/blob/1f152585232b919075189ea80eb42d531347609c/docs/Makefile#L4-L9
https://github.com/scikit-hep/pyhf/blob/1f152585232b919075189ea80eb42d531347609c/docs/Makefile#L54-L58

are just using `sphinx-build` anyway.

I'll make an Issue from this comment, but maybe @henryiii has additional thoughts here as well.